gum/backend-darwin/gumdarwinmodule.c

```diff
@@ -1400,6 +1475,150 @@ gum_darwin_module_load_image_from_filesystem (GumDarwinModule * self,
   return success;
 }
 
+static gboolean
+gum_darwin_module_load_image_header_from_filesystem (GumDarwinModule * self,
+                                                     const gchar * path,
+                                                     GumCpuType cpu_type,
+                                                     GError ** error)
+{
+  gboolean success;
+  GMappedFile * file;
+  gsize size, size_in_pages, page_size;
+  gpointer data;
+  GBytes * blob;
+  gsize header_offset = 0, header_size, cursor = 0;
```

Comment:
- `header_size` can be moved into the loop's scope.
- For clarity it would be good to initialize `cursor` just before the loop.

Reply:
uh, actually `header_size` is used also after the loop

Reply:
Oops, sorry, scratch that part 😄

Reply:
lol yeah it was `header_offset` instead 😅

Reply:
Yeah another case of me thinking one thing and saying something else 😂